src/main/scala/org/ergoplatform/nodeView/state/ErgoState.scala

```diff
@@ -105,21 +108,21 @@ object ErgoState extends ScorexLogging {
     val rewardOut = ByIndex(Outputs, IntConstant(0))
     val minerOut = ByIndex(Outputs, IntConstant(1))
 
-    val epoch = Plus(LongConstant(1), Divide(Minus(Height, LongConstant(s.fixedRatePeriod)), LongConstant(s.epochLength)))
-    val coinsToIssue = If(LT(Height, LongConstant(s.fixedRatePeriod)),
+    val epoch = Plus(IntConstant(1), Divide(Minus(Height, IntConstant(s.fixedRatePeriod.toInt)), IntConstant(s.epochLength)))
+    val coinsToIssue = If(LT(Height, IntConstant(s.fixedRatePeriod.toInt)),
       s.fixedRate,
-      Minus(s.fixedRate, Multiply(s.oneEpochReduction, epoch))
+      Minus(s.fixedRate.toInt, Multiply(s.oneEpochReduction.toInt, epoch))
```

Comment:
`fixedRate` and `oneEpochReduction` should remain Long

Reply:
@catena2w you mean the operations should be performed with long values? Because header is now Int.

Reply:
height is in, but this does not affect fixedRate and oneEpochReduction, they are still long

Reply:
@ergomorphic @catena2w Done. I've upcasted `epoch` to Long.